### PR TITLE
Refactor configuration of ClusterLogForwarder

### PR DESCRIPTION
**Pull request by DebakelOrakel**
With this commit configuring the ClusterLogForwarder resource will be done with a new parameter `openshift4_logging.clusterLogForwarder`, which will follow the upstream convention / spec.

The old parameter `openshift4_logging.clusterLogForwarding` is deprecated, but existing legacy config will be respected.

## Checklist

- [x] The PR has a meaningful title. It will be used to auto-generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [ ] Link this PR to related issues or PRs.

### CODE REVIEW

**Review by simu — APPROVED** [review]

File: tests/forwardingonly.yml [hunk 1]
```diff
@@ -28,6 +28,3 @@ parameters:
         enabled: false
       elasticsearch:
         enabled: false
-
-    clusterLogForwarding:
-      enabled: true
```
[the remark this answers is not in the document]
Comment (DebakelOrakel) [reply tests/forwardingonly.yml:33]: No, the default behaviour of the component was not to deploy a ClusterLogForwarder resource, when nothing is configured. And we have a set of clusters where the customer does manage the Forwarder resource himself. Not ideal and we should migrate to namespaced Forwarder there.

[the remark this answers is not in the document]
Comment (DebakelOrakel) [reply tests/forwardingonly.yml:33]: Its hard to have a default config, if you have a forwarding only configuration; what should be in there, where should it forward to.

[the remark this answers is not in the document]
Comment (simu) [reply tests/forwardingonly.yml:33]: Yes, but for the test case, you can invent anything, just take one of the examples from the docs

[the remark this answers is not in the document]
Comment (DebakelOrakel) [reply tests/forwardingonly.yml:33]: Oh, I have the testcases in the master test.

**Pull request merged**